src/odemis/driver/piezomotor.py

```diff
@@ -0,0 +1,871 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 18 Mar 2020
+
+@author: Philip Winkler
+
+Copyright © 2020, Philip Winkler, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+from __future__ import division
+
+import glob
+import logging
+from odemis import model
+from odemis.model import Actuator, CancellableThreadPoolExecutor
+from odemis.util import to_str_escape
+import os
+import threading
+import time
+import serial
+from threading import Thread
+import copy
+from odemis.model import HwError
+from odemis.util.driver import getSerialDriver
+
+BAUDRATE = 115200
+DEFAULT_AXIS_SPEED = 0.01  # m / s
+
+EOL = b'\r'  # 0x0D, carriage return
+sEOL = '\r'  # string version of EOL
+
+# Waveforms
+WAVEFORM_RHOMB = 1  # fast max speed
+WAVEFORM_DELTA = 2  # preferred, higher accuracy
+WAVEFORM_PARK = 4  # power off
+
+# Length of the rod available for moves
+STROKE_RANGE = 23.1e-3  # m
```

Comment:
This is the exact length of axis as specified in the datasheet of the motor. If we use a different kind of axis later, we should change it. I'm not sure if we're going to axes of different lengths. If not, it would be better to just put the precise value here, so we don't have to put the same value in all the yaml files. A move of 10 cm on the current axis will completely move the motor off the rod.